### PR TITLE
Creating src dir for System.diagnostics.tracing

### PR DESCRIPTION
**Pull request by dsgouda**
@weshaggard

### CODE REVIEW

**Comment by weshaggard** [comment]
@davmason @vancem please take a look.

File: src/Common/src/Interop/Windows/Interop.Libraries.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ internal static class Libraries
         internal const string Crypt32 = "crypt32.dll";
         internal const string Debug = "api-ms-win-core-debug-l1-1-0.dll";
         internal const string ErrorHandling = "api-ms-win-core-errorhandling-l1-1-0.dll";
+		internal const string Eventing = "api-ms-win-eventing-provider-l1-1-0.dll";
```
Comment (weshaggard) [inline src/Common/src/Interop/Windows/Interop.Libraries.cs]: indention.

File: src/Common/src/Interop/Windows/Interop.WinError.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+internal static partial class Interop
+{
```
Comment (weshaggard) [inline src/Common/src/Interop/Windows/Interop.WinError.cs]: These already exist https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Windows/mincore/Interop.Errors.cs

File: src/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.builds [hunk 3]
```diff
@@ -1,7 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
```
Comment (weshaggard) [inline src/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.builds]: Only have the TargetsWindows=true on the non empty Target groups.

File: src/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj [hunk 4]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <AssemblyName>System.Diagnostics.Tracing</AssemblyName>
+    <OutputType>Library</OutputType>
```
Comment (weshaggard) [inline src/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]: OutputType isn't necessary.

File: src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventDescriptor.cs [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+#endif
+
+#if ES_BUILD_STANDALONE
+namespace Microsoft.Diagnostics.Tracing
+#else
+namespace System.Diagnostics.Tracing
+#endif
+{
+    [StructLayout(LayoutKind.Explicit, Size = 16)]
+#if !netcore50
```
Comment (weshaggard) [inline src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventDescriptor.cs]: Can you do the initial commit of unchanged source codes and then a second commit with the changes you needed to make to the source code? That will allow us to easily see the diff of what you changed.

**Comment by vancem** [comment]
From the description, this is simply moving everything into a subdirectory.   Ideally you do nothing that is not needed for that and keep the delta very small otherwise. 

There are no semantic changes right?   

Note that this code is also shared with other enlistments, so keeping the delta to the actual data in the files is important.

File: src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventDescriptor.cs (hunk 5 above)
Comment (dsgouda) [inline src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventDescriptor.cs]: Created a separate commit to view modified src files

File: src/Common/src/Interop/Windows/Interop.WinError.cs (hunk 2 above)
Comment (weshaggard) [inline src/Common/src/Interop/Windows/Interop.WinError.cs]: It doesn't look like you converged these yet.

File: src/Common/src/Interop/Windows/Interop.Libraries.cs [hunk 6]
```diff
@@ -15,6 +15,7 @@ internal static class Libraries
...
```
Comment (weshaggard) [inline src/Common/src/Interop/Windows/Interop.Libraries.cs]: Looks like the indention is still off by on space here.

**Comment by weshaggard** [comment]
After looking at the code changes here I think you should undo them all including the define renames and file the issue for the owners to do the code clean-up.

File: src/.nuget/packages.Windows_NT.config [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.DotNet.BuildTools" version="1.0.25-prerelease-00134" />
```
Comment (weshaggard) [inline src/.nuget/packages.Windows_NT.config]: These are unrelated to your change.

**Comment by vancem** [comment]
Can I back up and understand the reason for the change?   

One thing that is not obvious is that unlike most packages in CoreFX, The EventSource code is SHARED by several source bases (Project N, CoreFX, and Desktop).   Thus we do NOT wish to change things in ways that make the (roughly by hand) integrations between these source bases needlessly hard. 

Thus small changes are OK, and a bulk move to a sub-directory is OK (but should be done as a single submission on its own because frankly doing comparisons across that commit will be significantly harder.   

Can you back up and tell me what it is you are trying to accomplish? (it seems to be more than just adding a src directory, which is what the title says).

**Comment by dsgouda** [comment]
@weshaggard please verify if this good to be posted as a PR against dotnet/master

File: dir.props [hunk 8]
```diff
@@ -11,7 +11,7 @@
 
   <!-- Build Tools Versions -->
   <PropertyGroup>
-    <BuildToolsVersion>1.0.25-prerelease-00134</BuildToolsVersion>
```
Comment (weshaggard) [inline dir.props]: This is unrelated to your change, please undo.

File: src/.nuget/packages.Unix.config [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.DotNet.BuildTools" version="1.0.25-prerelease-00134" />
```
Comment (weshaggard) [inline src/.nuget/packages.Unix.config]: Undo

File: src/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj [hunk 10]
```diff
@@ -1,25 +1,107 @@
...
-  </PropertyGroup>
-
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <AssemblyName>System.Diagnostics.Tracing</AssemblyName>
+    <ProjectGuid>{EB880FDC-326D-42B3-A3FD-0CD3BA29A7F4}</ProjectGuid>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <AssemblyVersion>4.0.21.0</AssemblyVersion>
+    <!-- Need to remove reference to System.Reflection.TypeExtensions
```
Comment (weshaggard) [inline src/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]: This comment should be moved to OmitTransitiveCompileReferences.

**Comment by dsgouda** [comment]
Addressed all comments, opening new PR at dotnet/master
